components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -330,7 +330,8 @@ func reaper(terminatingReaper <-chan bool) {
 			continue
 		}
 
-		pid, err := unix.Wait4(-1, nil, 0, nil)
+		// "pid: 0, options: 0" to follow https://github.com/ramr/go-reaper/issues/11 to make agent-smith work again
```

Comment:
This needs some more explanation:
1. passing `pid: 0` to `wait4` makes it behave like passing `pid: 0` to [`waitpid`](https://man7.org/linux/man-pages/man2/waitpid.2.html) which means we're just reaping child processes within the same process group. The comment should reflect that.
2. I don't quite understand how this makes agent smith work again? What's its the relationship to the reaping? I'd also be happy to discuss this offline.

Reply:
> I'd also be happy to discuss this offline.
Absolutely. I'm not sure of all effects either, makes 100% sense to discuss in-depth.

Reply:
We agreed that we do not understand 100% why the reaper receives signals from the pod PID namespace at all. But also the negative impact seems begnin, so we're going with this.